service/src/main/java/com/myhome/services/springdatajpa/AmenitySDJpaService.java

```diff
@@ -114,4 +120,20 @@ public boolean deleteBooking(String bookingId) {
                 })
                 .orElse(false);
     }
+
+    @Override
+    public Optional<List<AmenityBookingItem>> listAllAmenityBookings(String amenityId,
+        LocalDate startDate,
+        LocalDate endDate,
+        Pageable pageable) {
+          LocalDateTime start = startDate != null ? startDate.atStartOfDay() : null;
+          LocalDateTime end = endDate != null ? endDate.atStartOfDay() : null;
+          
+      List<AmenityBookingItem> bookingItems =
+          bookingRepository.findAllByAmenity(amenityId, start, end, pageable);
+      if (bookingItems.isEmpty() && !amenityRepository.findByAmenityId(amenityId).isPresent()) {
```

Comment:
Maybe it would be better to switch the conditions?
First call `findByAmenityId` and then if present call `findAllByAmenity`

Reply:
you're right, thanks